django/overwolf_auth/backends.py

```diff
@@ -0,0 +1,147 @@
+from django.urls import reverse
+from jwt import PyJWKClient
+from jwt import decode as jwt_decode
+from social_core.backends.oauth import BaseOAuth2  # type: ignore
+from social_core.exceptions import AuthException  # type: ignore
+
+Response = Dict[str, Any]
+
+
+class UserDetails(TypedDict):
+    username: str
+    email: str
+    fullname: str
+    first_name: str
+    last_name: str
+
+
+class OverwolfOAuth2(BaseOAuth2):
+    """
+    Overwolf OAuth 2 backend
+
+    https://overwolf.github.io/topics/integrations/login-with-overwolf
+    """
+
+    OW_URL = "https://accounts.overwolf.com/oauth2"
+
+    # Overwrite values from BaseAuth
+    name = "overwolf"
+    # supports_inactive_user = False  # Django auth
+    # EXTRA_DATA = None
+    # GET_ALL_EXTRA_DATA = False
+    # REQUIRES_EMAIL_VALIDATION = False
+    # SEND_USER_AGENT = False
+    # SSL_PROTOCOL = None
+
+    # Overwrite values from OAuthAuth
+    ACCESS_TOKEN_METHOD = "POST"
+    ACCESS_TOKEN_URL = f"{OW_URL}/token"
+    AUTHORIZATION_URL = f"{OW_URL}/auth"
+    # DEFAULT_SCOPE = None
+    ID_KEY: Literal["username"] = "username"  # Default was "id"
+    # REVOKE_TOKEN_METHOD = "POST"
+    # REVOKE_TOKEN_URL = None
+    # SCOPE_PARAMETER_NAME = "scope"
+    # SCOPE_SEPARATOR = " "
+
+    # Overwrite values from BaseOAuth2
+    REDIRECT_STATE = False  # Disable state validation
+    # REFRESH_TOKEN_METHOD = "POST"
+    # REFRESH_TOKEN_URL = None
+    # RESPONSE_TYPE = "code"
+    STATE_PARAMETER = False  # Disable state validation
+
+    def extra_data(
+        self,
+        user: Any,
+        uid: Any,
+        response: Response,
+        details: UserDetails = None,
+        *args,
+        **kwargs,
+    ) -> Dict[str, Any]:
+        """
+        Override implementation from BaseOAuth2.
+
+        Augment the info stored in "extra_data" database field with info
+        from the JWT. The JWT itself is also stored, but not needing to
+        decode the token everywhere the info is used makes sense.
+        """
+        data = super().extra_data(user, uid, response, details=details, *args, **kwargs)
+
+        jwt_data = self._decode_response_jwt(response)
+        data["email_verified"] = jwt_data.get("email_verified")
+        data["nickname"] = jwt_data.get("nickname")
+        data["picture"] = jwt_data.get("picture")
+        data["preferred_username"] = jwt_data.get("preferred_username")
+
+        return data
+
+    def get_redirect_uri(self, state: Optional[Any] = None) -> str:
+        """
+        Override implementation from OAuthAuth.
+        """
+        host = (
+            settings.SOCIAL_AUTH_INIT_HOST
+            or settings.AUTH_EXCLUSIVE_HOST
+            or settings.PRIMARY_HOST
+        )
+
+        if host is None:
+            raise ImproperlyConfigured("No host for OverwolfOAuth2")
+
+        path = reverse("social:complete", kwargs={"backend": self.name})
+        return f"{settings.PROTOCOL}{host}{path}"
```

Comment:
I think this function isn't needed as the other auth methods don't need it either.

If I remember correctly, the auth is _initiated_ from the auth domain, as the login buttons link there. It's just that the auth complete redirect URL redirects away from it, but that's handled via URL params.

So I'd make sure this function is actually necessary, I have a strong feeling it might not be